packages/vue-renderer/src/renderer.js

```diff
@@ -141,34 +141,28 @@ export default class VueRenderer {
     }
   }
 
-  loadResources(_fs, isMFS = false) {
+  async loadResources(_fs) {
     const distPath = path.resolve(this.context.options.buildDir, 'dist', 'server')
     const updated = []
-    const { resourceMap } = this
 
-    const readResource = (fileName, encoding) => {
+    const readResource = async (fileName, encoding) => {
       try {
         const fullPath = path.resolve(distPath, fileName)
-        if (!_fs.existsSync(fullPath)) {
+        if (!await _fs.exists(fullPath)) {
           return
         }
-        const contents = _fs.readFileSync(fullPath, encoding)
```

Comment:
No need cleaning ?

Reply:
Cleaning only (potentially) saves a few KBs (Bigger projects, few MBs) of RAM. But adds the complexity of oldAssets. Also if we want to support independent server/webpack restarts we need correct MFS snapshot.

I thought the cleaner code with fewer hacks worth it. What do you think?

Reply:
I agree for most resources.

Only `_assetsMapping`, if we don't clear the old object, it may have wrong mappings after rebuilding.

Reply:
Oh you are right. We have to invalidate that mapping when loading assets. Does it look good?

https://github.com/nuxt/nuxt.js/pull/5186/files#diff-015dff3d45b52153ef5f24ef104e954bR189

Reply:
Great, I always love your code, bro, neat refactoring!